### PR TITLE
WIP - some changes to the serializers.

### PR DESCRIPTION
**Pull request by jasongrout**
This was originally part of #1294, and was reverted in c64e5f67e3fa3e75e6de209bc1a644034bf7e122

### CODE REVIEW

**Comment by jasongrout** [comment]
The get_state changes are incorporated in #1302.

**Comment by jasongrout** [comment]
I just rebased keeping the relevant changes from before.

File: jupyter-js-widgets/src/widget.ts [hunk 1]
```diff
@@ -391,31 +391,34 @@ class WidgetModel extends Backbone.Model {
...
                 }
-                if (state[k].toJSON) {
-                    state[k] = state[k].toJSON();
+                if (newState[k].toJSON) {
+                    newState[k] = newState[k].toJSON();
                 }
             } catch (e) {
-                console.error("Error serializing widget state attribute: ", k);
+                console.error(`Error serializing widget state attribute ${k} (value ${state[k]})`);
                 throw e;
```
Comment (maartenbreddels) [inline jupyter-js-widgets/src/widget.ts]: I believe IE doesnt support this, right @SylvainCorlay, what's it called, template strings orso?

Comment (jasongrout) [inline jupyter-js-widgets/src/widget.ts]: typescript compiles this down to normal js strings.

Comment (jasongrout) [inline jupyter-js-widgets/src/widget.ts]: (example: https://www.typescriptlang.org/play/#src=class%20A%3CT%3E%20%7B%0A%20%20%20%20x%20%3D%20new%20T()%3B%0A%7D%0A#src=let%20x%20%3D%205%0Alet%20y%20%3D%20%60Test%20%24%7Bx%7D%3B%20it%20works!%60%0A  )

Comment (maartenbreddels) [inline jupyter-js-widgets/src/widget.ts]: 👍

**Comment by jasongrout** [comment]
Closed in favor of #1355, which has the last part of the above.

**Pull request closed without merging**